### ISSUE TITLE
SearchBox with ES 7.0

### ISSUE THREAD

**Issue by cb29** [searchkit/searchkit#717]
Hello,
I'm using SearchKit on an ES cluster 7.0, and I have encountered several problems: HitsStats, Pagination...
I'm having a problem with the SearchBox component. The search don't work. When I use the SearchBox, I get an error 400.

Has anyone ever used SearchBox with ES 7.0? If yes can you give me an example of configuration?

Here is my SearchBox config 
`          <SearchBox
            searchOnChange={false}
            queryOptions={{analyzer:"fuzziness"}}        
	    queryFields={["title","field_related_make_name"]}
	    prefixQueryFields={["title","field_related_make_name"]}
	    />`

Thanks for your help.

**Comment by joemcelroy** [comment]
hi there, could i see the error reason that ES7 returns.

**Comment by pahan35** [comment]
There is an error due to changes in `results.hits.total`  https://www.elastic.co/guide/en/elasticsearch/reference/7.0/search-request-track-total-hits.html
https://www.elastic.co/guide/en/elasticsearch/reference/master/search-request-body.html#request-body-search-track-total-hits

As I remember, you need to fix `getHitsCount()` and `hasHits()` inside `SearchkitManager`

My overriding fix
```
class SearchkitManager7 extends SearchkitManager {
  getHitsCount() {
    return get(this.results, ['hits', 'total'], {
      relation: 'eq',
      value: 0,
    })
  }

  hasHits() {
    const {value} = this.getHitsCount()
    return value > 0
  }
}
```

I guess you need to make it back compatible

**Comment by pahan35** [comment]
Also, #697 is related

**Comment by joemcelroy** [comment]
nice workaround! thanks @pahan35 - need to think about how to compat for 6 and 7.

**Comment by joemcelroy** [comment]
fixed in v2.4.1
